src/types/onyx/Policy.ts

```diff
@@ -68,18 +69,115 @@ type TaxRatesWithDefault = {
-// Once the data is being used, these types should be defined appropriately.
-type ConnectionLastSync = Record<string, unknown>;
-type ConnectionData = Record<string, unknown>;
-type ConnectionConfig = Record<string, unknown>;
-type Connection = {
+type ConnectionLastSync = {
+    successfulDate?: string;
+    errorDate?: string;
+    isSuccessful: boolean;
+    source: 'DIRECT' | 'EXPENSIFYWEB' | 'EXPENSIFYAPI' | 'AUTOSYNC' | 'AUTOAPPROVE';
+};
+
+type Account = {
+    glCode?: string;
+    name: string;
+    currency: string;
+    id: string;
+};
+
+type Employee = {
+    id: string;
+    firstName?: string;
+    lastName?: string;
+    name: string;
+    email: string;
+};
+
+type Vendor = {
+    id: string;
+    name: string;
+    currency: string;
+    email: string;
+};
+
+type TaxCode = {
+    totalTaxRateVal: string;
+    simpleName: string;
+    taxCodeRef: string;
+    taxRateRefs: Record<string, string>;
+    name: string;
+};
+
+type QuickBooksOnlineConnectionData = {
+    country: string;
+    edition: string;
+    homeCurrency: string;
+    isMultiCurrencyEnabled: boolean;
+
+    journalEntryAccounts: Account[];
+    bankAccounts: Account[];
+    creditCards: Account[];
+    accountsReceivable: Account[];
+    accountsPayable: Account[];
+    otherCurrentAssetAccounts: Account[];
+
+    taxCodes: TaxCode[];
+    employees: Employee[];
+    vendors: Vendor[];
+};
+
+type IntegrationEntityMap = 'NONE' | 'DEFAULT' | 'TAG' | 'REPORT_FIELD';
+
+type ConnectionConfig = {
+    realmId: string;
+    credentials: {
+        access_token: string;
+        refresh_token: string;
+        expires: number;
+        companyID: string;
+        readlmId: string;
+        scope: string;
+        companyName: string;
+        token_type: string;
+    };
+    companyName: string;
+    autoSync: {
+        jobID: string;
+        enabled: boolean;
+    };
+    syncPeople: boolean;
+    syncItems: boolean;
+    markChecksToBePrinted: boolean;
+    reimbursableExpensesExportDestination: IntegrationEntityMap;
+    nonReimbursableExpensesExportDestination: IntegrationEntityMap;
+
+    reimbursableExpensesAccount?: string;
+    nonReimbursableExpensesAccount?: string;
+    autoCreateVendor: boolean;
+    hasChosenAutoSyncOption: boolean;
+    syncClasses: IntegrationEntityMap;
+    syncCustomers: IntegrationEntityMap;
+    syncLocations: IntegrationEntityMap;
+    exportDate: string;
+    lastConfigurationTime: number;
+    syncTax: boolean;
+    enableNewCategories: boolean;
+    export: {
+        exporter: string;
+    };
+};
+type Connection<ConnectionData> = {
     lastSync?: ConnectionLastSync;
     data: ConnectionData;
     config: ConnectionConfig;
 };
 
+type Connections = {
+    quickbooksOnline: {
+        lastSync?: ConnectionLastSync;
+        data: Connection<QuickBooksOnlineConnectionData>;
+        config: ConnectionConfig;
+    };
```

Comment:
This doesn't make sense to me now, if `Connection` is already defined as `{lastSync, data, config}`, then this would mean that the `quickbooksOnline` object would be:
```
lastSync,
data: {
    lastSync
    data:
    config
},
config
```
?

Looks like this should be this instead:
```suggestion
    quickbooksOnline: Connection<QuickBooksOnlineConnectionData>;
```

Reply:
yes that's right!